metal/loadbalancers/emlb/emlb.go

```diff
@@ -10,24 +10,62 @@ import (
 )
 
 type LB struct {
-	client *lbaas.APIClient
+	client     *lbaas.APIClient
+	controller *controller
 }
 
 func NewLB(k8sclient kubernetes.Interface, config string) *LB {
+	metalAPIKey := "TODO"
 	lb := &LB{}
-	emlbConfig := lbaas.NewConfiguration()
-	lb.client = lbaas.NewAPIClient(emlbConfig)
+	lb.controller = NewController(metalAPIKey)
 	return lb
 }
 
 func (l *LB) AddService(ctx context.Context, svcNamespace, svcName, ip string, nodes []loadbalancers.Node) error {
+	// 1. Gather the properties we need: Metal API key, port number(s), cluster name(?), target IP(s?)
+	additionalProperties := map[string]string{}
+
+	// 2. Create the infrastructure (what do we need to return here?  lb name and/or ID? anything else?)
```

Comment:
If the LB API requires a Metal IP or IP Reservation (rather than dynamically provisioning one), CPEM may need to provision that IP Reservation and annotate (Metal API tag) it in such a way that it can be identified and reconciled (deleted from the Metal API) when the K8s LB Service is deleted. 

We have this workflow in place elsewhere in this controller, perhaps it is already abstracted above the MetalLB/KubeVip/Other implementation layer.

Reply:
Looks to be handled in: https://github.com/kubernetes-sigs/cloud-provider-equinix-metal/blob/main/metal/eip_controlplane_reconciliation.go

But perhaps there may be some need to translate a Metal IP to something LBaaS recognizes?

Reply:
As a first pass we're focused on service load balancing because the control plane load balancing is currently less flexible in CPEM:  for control plane, we don't want to put a load balancer in front of an EIP that flaps from node to node, we want to use a load balancer instead of an EIP.

Reply:
Currently LB allocates the IP for you, so we don't need to track or allocate this ourselves.

Reply:
That may be actually require more code. It looks like the EIPs are currently passed in to each implementation (having already been determined through the Metal API).